viz/cmd/tap.go

```diff
@@ -428,15 +461,60 @@ func renderTapEvent(event *tapPb.TapEvent) string {
 }
 
 // renderTapEventJSON renders a Public API TapEvent to a string in JSON format.
-func renderTapEventJSON(event *tapPb.TapEvent) string {
+func renderTapEventJSON(event *tapPb.TapEvent, opts ...renderOptions) string {
+	filter := &renderFilter{}
+	for _, opt := range opts {
+		opt(filter)
+	}
 	m := mapPublicToDisplayTapEvent(event)
+	if filter.JsonPath != "" {
+		filteredJson, err := getJsonFilteredByJPath(m, filter.JsonPath)
+		if err != nil {
+			return err.Error()
+		}
+		return filteredJson[0]
+	}
 	e, err := json.MarshalIndent(m, "", "  ")
 	if err != nil {
 		return fmt.Sprintf("{\"error marshalling JSON\": \"%s\"}", err)
 	}
 	return string(e)
 }
 
+func getJsonFilteredByJPath(event interface{}, jsonPath string) ([]string, error) {
+	fields, err := get.RelaxedJSONPathExpression(jsonPath)
+	if err != nil {
+		return []string{}, err
+	}
+
+	j := jsonpath.New("EventParser")
+	if err := j.Parse(fields); err != nil {
+		return []string{}, err
+	}
+
+	results, err := j.FindResults(event)
+	if err != nil {
+		return []string{}, err
+	}
+
+	filteredEvent := []string{}
+	if len(results) == 0 || len(results[0]) == 0 {
+		return filteredEvent, errors.New("{\"error filtering JSON\": \"couldn't find any results matching with jsonpath filter\"}")
+	}
+
+	for _, result := range results {
+		for _, match := range result {
+			e, err := json.MarshalIndent(match.Interface(), "", "  ")
+			if err != nil {
+				log.Panicf("Error %v", err.Error())
```

Comment:
Why panicking instead of returning an error?

Reply:
Done.